modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/data_set/filters/modals/selection_filter/source_type/ApiRestApplication.tsx

```diff
@@ -0,0 +1,671 @@
+import RESTApplicationDropdownMenu from '../../../../../components/rest/RESTApplicationDropdownMenu';
+import RESTEndpointDropdownMenu from '../../../../../components/rest/RESTEndpointDropdownMenu';
+import RESTSchemaDropdownMenu from '../../../../../components/rest/RESTSchemaDropdownMenu';
+import {
+	ALLOWED_ENDPOINTS_PARAMETERS,
+	FUZZY_OPTIONS,
+} from '../../../../../utils/constants';
+import getFields from '../../../../../utils/getFields';
+import openDefaultFailureToast from '../../../../../utils/openDefaultFailureToast';
+import {IField, ISelectionFilter} from '../../../../../utils/types';
+
+interface IApiRestApplicationModalContentProps {
+	filter?: ISelectionFilter;
+	itemKeyValidationError: boolean;
+	itemLabelValidationError: boolean;
+	namespace: string;
+	onChange: ({
+		selectedItemKey,
+		selectedItemLabel,
+		selectedRESTApplication,
+		selectedRESTEndpoint,
+		selectedRESTSchema,
+	}: {
+		selectedItemKey: string;
+		selectedItemLabel: string;
+		selectedRESTApplication: string | null;
+		selectedRESTEndpoint: string | null;
+		selectedRESTSchema: string | null;
+	}) => void;
+	requiredRESTApplicationValidationError: boolean;
+	restApplications: string[];
+	restEndpointValidationError: boolean;
+	restSchemaValidationError: boolean;
+}
+
+function ApiRestApplication({
+	filter,
+	itemKeyValidationError,
+	itemLabelValidationError,
+	namespace,
+	onChange,
+	requiredRESTApplicationValidationError,
+	restApplications,
+	restEndpointValidationError,
+	restSchemaValidationError,
+}: IApiRestApplicationModalContentProps) {
+	const [fields, setFields] = useState<IField[]>([]);
+	const [selectedItemKey, setSelectedItemKey] = useState<string>(
+		filter?.itemKey ? filter.itemKey : ''
+	);
+	const [selectedItemLabel, setSelectedItemLabel] = useState<string>(
+		filter?.itemLabel ? filter.itemLabel : ''
+	);
+	const [
+		noEnpointsRESTApplicationValidationError,
+		setNoEnpointsRESTApplicationValidationError,
+	] = useState(false);
+	const [restSchemaEndpoints, setRESTSchemaEndpoints] = useState<
+		Map<string, Array<string>>
+	>(new Map());
+	const [selectedRESTApplication, setSelectedRESTApplication] = useState<
+		string | null
+	>(filter?.restApplication ? filter.restApplication : null);
+	const [selectedRESTSchema, setSelectedRESTSchema] = useState<string | null>(
+		filter?.restSchema ? filter.restSchema : null
+	);
+	const [selectedRESTEndpoint, setSelectedRESTEndpoint] = useState<
+		string | null
+	>(filter?.restEndpoint ? filter.restEndpoint : null);
+
+	const isPathValid = (
+		path: string,
+		allowedParameters: string[]
+	): boolean => {
+		const paramsMatcher = RegExp('{(.*?)}', 'g');
+		let matches;
+
+		while ((matches = paramsMatcher.exec(path)) !== null) {
+			if (!allowedParameters.includes(matches[1])) {
+				return false;
+			}
+		}
+
+		return true;
+	};
+
+	const getRESTSchemas = async (restApplication: string) => {
```

Comment:
There is still quite of lot of duplication here. We can probably extract `getRESTSchemas` to a util, just like we do with `getFields`. But, we can do this in a followup. This PR is cooking long enough :slightly_smiling_face: